test/unit/multiplication/test_multiplication_general.cpp

```diff
@@ -29,16 +31,114 @@ using namespace dlaf;
 template <class T>
 struct GeneralMultiplicationTestGPU : public ::testing::Test {};
-
 TYPED_TEST_SUITE(GeneralMultiplicationTestGPU, MatrixElementTypes);
+
+template <class T>
+struct GeneralSubMultiplicationTestGPU : public ::testing::Test {};
+TYPED_TEST_SUITE(GeneralSubMultiplicationTestGPU, MatrixElementTypes);
+
+template <class T>
+struct GeneralSubMultiplicationDistTestGPU : public TestWithCommGrids {};
+TYPED_TEST_SUITE(GeneralSubMultiplicationDistTestGPU, MatrixElementTypes);
+#endif
+
+struct GemmConfig {
+  const blas::Op opA;
+  const blas::Op opB;
+
+  const SizeType m;
+  const SizeType n;
+  const SizeType k;
+
+  const SizeType mb;
+  const SizeType nb;
+  const SizeType kb;
+};
+
+template <class T, Backend B, Device D>
+void testGeneralMultiplication(const T alpha, const T beta, const GemmConfig& config) {
+  using dlaf::matrix::internal::MatrixRef;
+
+  auto [refA, refB, refC, refResult] =
+      matrix::test::getMatrixMatrixMultiplication<GlobalElementIndex, T>(config.opA, config.opB,
+                                                                         config.k, alpha, beta);
+
+  auto setMatrix = [&](auto elSetter, const LocalElementSize size, const TileElementSize block_size) {
+    Matrix<T, Device::CPU> matrix(size, block_size);
+    dlaf::matrix::util::set(matrix, elSetter);
+    return matrix;
+  };
+
+  Matrix<const T, Device::CPU> mat_ah = setMatrix(refA, {config.m, config.k}, {config.mb, config.kb});
+  Matrix<const T, Device::CPU> mat_bh = setMatrix(refB, {config.k, config.n}, {config.kb, config.nb});
+  Matrix<T, Device::CPU> mat_ch = setMatrix(refC, {config.m, config.n}, {config.mb, config.nb});
+
+  {
+    MatrixMirror<const T, D, Device::CPU> mat_a(mat_ah);
+    MatrixMirror<const T, D, Device::CPU> mat_b(mat_bh);
+    MatrixMirror<T, D, Device::CPU> mat_c(mat_ch);
+
+    MatrixRef<const T, D> mat_sub_a(mat_a.get(), {{0, 0}, mat_ah.size()});
+    MatrixRef<const T, D> mat_sub_b(mat_b.get(), {{0, 0}, mat_bh.size()});
+    MatrixRef<T, D> mat_sub_c(mat_c.get(), {{0, 0}, mat_ch.size()});
+
+    // Note: currently it is implemented just the NoTrans/NoTrans case
+    ASSERT_EQ(config.opA, blas::Op::NoTrans);
+    ASSERT_EQ(config.opB, blas::Op::NoTrans);
+    multiplication::internal::General<B, D, T>::callNN(alpha, mat_sub_a, mat_sub_b, beta, mat_sub_c);
+  }
+
+  CHECK_MATRIX_NEAR(refResult, mat_ch, 40 * (mat_ch.size().rows() + 1) * TypeUtilities<T>::error,
+                    40 * (mat_ch.size().rows() + 1) * TypeUtilities<T>::error);
+}
+
+std::vector<GemmConfig> full_gemm_configs = {
+    // empty matrices
+    GemmConfig{blas::Op::NoTrans, blas::Op::NoTrans, 0, 0, 7, 3, 6, 2},
+    GemmConfig{blas::Op::NoTrans, blas::Op::NoTrans, 20, 0, 7, 3, 6, 2},
+    GemmConfig{blas::Op::NoTrans, blas::Op::NoTrans, 0, 20, 7, 3, 6, 2},
+
+    // full
+    GemmConfig{blas::Op::NoTrans, blas::Op::NoTrans, 21, 21, 21, 3, 3, 3},
+    GemmConfig{blas::Op::NoTrans, blas::Op::NoTrans, 12, 20, 11, 3, 4, 5},
```

Comment:
Add other test cases. E.g.:
- One of the matrices has a single incomplete tile
- a couple of other test with a full matrix